pkg/exchange/max/exchange.go

```diff
@@ -652,6 +652,20 @@ func (e *Exchange) SubmitOrder(ctx context.Context, order types.SubmitOrder) (cr
 			)
 
 			if responseErr.StatusCode >= 400 {
+				if responseErr.IsJSON() {
+					var maxError maxapi.ErrorResponse
+					if err := responseErr.DecodeJSON(&maxError); err != nil {
+						logger.WithError(err).Errorf("unable to decode max api error response: %s", responseErr.Response.Body)
```

Comment:
Should we return here to avoid entering following switch case?

Reply:
You're right, let me do this